src/main/java/io/debezium/connector/yugabytedb/YugabyteDBConnector.java

```diff
@@ -196,6 +205,14 @@ private YBClient getYBClientBase(String hostAddress, long adminTimeout, long ope
 
     @Override
     public void stop() {
+        LOGGER.info("Stopping table monitoring thread");
+        tableMonitorThread.shutdown();
+        try {
+            tableMonitorThread.join(MAX_TIMEOUT);
+        } catch (InterruptedException ie) {
+            // Ignore, shouldn't be interrupted.
```

Comment:
Can you put a warning message just as a fall back that it may not have worked.

Reply:
Done.